internal/broker/broker.go

```diff
@@ -102,6 +107,13 @@ type Option func(*option)
 func New(cfg Config, args ...Option) (b *Broker, err error) {
 	defer decorate.OnError(&err, "could not create broker")
 
+	if cfg.ConfigPath != "" {
+		err = parseConfig(cfg.ConfigPath, &cfg)
+		if err != nil {
+			return nil, fmt.Errorf("could not parse config: %v", err)
+		}
+	}
```

Comment:
I don't think having two structs `broker.Config` and a `broker.brokerConfig` is a good idea. I prefer using a single struct if they don't serve a different purpose (which the difficulty to find different names shows). I don't see a problem with encapsulation, `parseConfig` is an unexported function which sets unexported fields of the struct. There is no object here which sees something it shouldn't see.

Reply:
Sure, but it should take a path as an argument and return a parsed configuration rather than filling a pointer.

Reply:
>  but it should take a path as an argument 

It already does.

> and return a parsed configuration rather than filling a pointer

Not if it requires making the code more complex, which it does AFAICT. I don't understand the strong opposition to passing a pointer and letting the called function modify the fields when the function signature makes it clear that that's what happens.

Reply:
My issue with this is mainly that this is more of a C-like approach to a function than a Go one. It also depends on the decision we take on the previous comment regarding using different structs. But, like I've said, it's more straightforward and Go-like to have a function that takes the path as an argument and returns a concrete type with the expected content. You can see this type of difference when looking at `os.ReadFile`, for example. In C, it is common to allocate the buffer and then read something into it, but in Go we tend to return what was read instead.

I'm not sure if I was able to explain it as clear as possible. Did it make sense? We can talk on MM also if you're not satisfied.

Reply:
`parseConfigFile` now returns a `userConfig` struct.